pkg/smb/nodeserver.go

```diff
@@ -501,15 +513,74 @@ func ensureKerberosCache(mountFlags []string, secrets map[string]string) (bool,
 		if err != nil {
 			return false, err
 		}
-		err = os.WriteFile(krb5CacheFileName, content, os.FileMode(0700))
+		// Write cache into volumeId-based filename, so it can be cleaned up later
+		volumeIDCacheFileName := volumeKerberosCacheName(volumeID)
+
+		var volumeIDCacheAbsolutePath = getKerberosFilePath(volumeIDCacheFileName)
+		err = os.WriteFile(volumeIDCacheAbsolutePath, content, os.FileMode(0700))
+		if err != nil {
+			return false, status.Error(codes.Internal, fmt.Sprintf("Couldn't write kerberos cache to file %s: %v", volumeIDCacheAbsolutePath, err))
+		}
+		err = os.Chown(volumeIDCacheAbsolutePath, credUID, credUID)
 		if err != nil {
-			return false, status.Error(codes.Internal, fmt.Sprintf("Couldn't write kerberos cache to file %s: %v", krb5CacheFileName, err))
+			return false, status.Error(codes.Internal, fmt.Sprintf("Couldn't chown kerberos cache %s to user %d: %v", volumeIDCacheAbsolutePath, credUID, err))
 		}
-		err = os.Chown(krb5CacheFileName, credUID, credUID)
+		// Delete target file if exists, it's safe to ignore errors.
+		_ = os.Remove(krb5CacheFileName)
+
+		// Create symlink to the cache file with expected name
+		err = os.Symlink(volumeIDCacheAbsolutePath, krb5CacheFileName)
```

Comment:
write like this golang style:
```
if err := ...; err != nil {

}
```

Reply:
why create a symlink here?

Reply:
I've put up explanation in method doc, basically unstage provides very little information, specifically secret is missing, so the name of the file can't be found. Using filename based on the volumeID allows to do a cleanup, having symlink provides the expected file to mount process